### ISSUE TITLE
Print to PDF does not capture all output

### ISSUE THREAD

**Issue by NeapolitanSuitcase** [googlecolab/colabtools#4357]
**Describe the current behavior**

The bottom few pages of the notebook+output are not included in the printed PDF. It cuts off a little ways before the end.

Possible reason: print option may not look for enough pages- possibly not taking output into account or similar.

**Describe the expected behavior**

"Print to PDF" will print the contents of all cells as well as their outputs, including the last cell/output on the page.

**What web browser you are using**

Chrome

**Additional context**

https://colab.research.google.com/drive/16jN-bscNaHoZZ5hizApbolkAvPkuaMmY?usp=sharing

Note: This is not a very minimal file. Suspect issue is more obvious with longer files.

**Comment by NeapolitanSuitcase** [comment]
[Public Print to PDF copy.pdf](https://github.com/googlecolab/colabtools/files/14160921/Public.Print.to.PDF.copy.pdf)

Attached is the print to PDF output. Does not match the file itself.

**Comment by wwic7oria** [comment]
I have the same issue. I was trying to export as pdf my project for university, but it doesn't show properly the end of the notebook file. I was trying to do something with this and noticed, that if I change the margin of output pdf file, it also changes how the end of that pdf file looks

Notebook file:
https://colab.research.google.com/drive/1s1hgozSQUjqdxJ0VUmoJk407TG1o88dw?usp=sharing

PDF with default margin:
[pdf on google drive](https://drive.google.com/file/d/1BoqSho66RbrSZLbcJcFNK5VywN2NCJWf/view?usp=sharing) or [download from github](https://github.com/googlecolab/colabtools/files/15337869/default.margin.pdf)

PDF with min margin:
[pdf on google drive](https://drive.google.com/file/d/1d8Q1JuKfZNOlwG740aAR4XN3Es9O06Pv/view?usp=sharing) or 
[download from github](https://github.com/googlecolab/colabtools/files/15337866/min.margin.pdf)

As you can see, the more margin is - the more text is at the end of the pdf file. But I think it's a random behavior, because increasing margin doesn't always work

**Comment by FunRoss7** [comment]
Yeah, I have this problem too. I can only print 10 pages. As a workaround I will split my project and print each section separately.

**Comment by olekniewiarowski** [comment]
I have the same issue. Output is truncated at 14 pages. Non-default settings: {default page layout: single}

**Comment by olekniewiarowski** [comment]
Thanks for the suggestion - I tried on different browsers - each browser results in a different length output (in terms of scope, not number of pages). Still no luck in getting my full document out. 

Also tweaking scale changes how much document scope makes it into the print. Weird.

**Comment by BethTian** [comment]
I have a good solution, just creat more blank code windows, then you can print more contents until it contains all of the codes we want to print

**Comment by KhadijahAgboola** [comment]
> I have a good solution, just creat more blank code windows, then you can print more contents until it contains all of the codes we want to print

Please explain more on this

**Comment by RuzyG** [comment]
> I have a good solution, just creat more blank code windows, then you can print more contents until it contains all of the codes we want to print

Really good method!! THANKS!!!

**Comment by TimothyWng** [comment]
A solution I found was to open the notebook in Microsoft Edge, then print using File -> Print, then set custom pages from 1-### whatever page includes all your output. I'm not sure if this works for every case, but it worked for my 13 page output.

**Comment by spencershumwaygoogle** [comment]
Thanks for sharing workarounds @TimothyWng and @BethTian. I have also filed an internal bug to track this b/394660125.

**Comment by taoy66** [comment]
For Chrome, to print to PDF with all output, follow this:

File > Print > More Settings > Change Scale to 'Custom', then adjust the number accordingly

**Comment by weebao** [comment]
Any solution for this yet. I'm trying to print an 18-page notebook but it keeps getting cut off at 16. Tried all the solution above but didn;'t work

**Comment by mochi52** [comment]
I tried to print a 26-pages document on Chrome, couldn't get it to print all pages, maximum it could do was 20 pages. Tried again using Safari and it successfully printed full 26 pages with all outputs.

**Comment by jjv432** [comment]
I was having an issue with matplotlib.pyplot plots. The bottom of one plot was 'hidden' behind the text of a code section. I was able to fix it by doing the following: 

`!pip install --pre -U matplotlib`

source: https://stackoverflow.com/questions/58791437/how-to-fix-google-colab-cutting-of-plots-of-confusion-matrices

**Comment by PrashantMaht0** [comment]
> For Chrome, to print to PDF with all output, follow this:
> 
> File > Print > More Settings > Change Scale to 'Custom', then adjust the number accordingly

This Thing worked prefectly for me, I mean the code look smaller than it should but at least the output cells are properly formatted

**Comment by avionerman** [comment]
Choose the result with your mouse (bottom to top), right-click -> print, and then you will see in a PDF (only!) the cell result - from there you can download it easily without experimenting with options from the 'colab print pdf capability'.

**Comment by bethzianne** [comment]
> I have a good solution, just creat more blank code windows, then you can print more contents until it contains all of the codes we want to print

This fixed it for me, thank you so much
